### ISSUE TITLE
GoTest failing on neovim when g:go_term_enabled undefined or zero

### ISSUE THREAD

**Issue by hoop33** [fatih/vim-go#1446]
### Behavior

When running `:GoTest` in neovim when `g:go_term_enabled` is 0 or undefined, so that the job functionality is used, I get:

```
vim-go: testing ...
Error detected while processing function go#test#Test[56]..go#jobcontrol#Spawn[4]..<SNR>187_spawn:
line   47:
E745: Using a List as a Number
E15: Invalid expression: ['go'] + a:args
```

### Steps to reproduce:

1. Set `g:go_term_enabled` to 0 (or leave undefined)
2. Open a go test file
3. Run `:GoTest`

### Thoughts

This seems to have been caused by https://github.com/fatih/vim-go/commit/df60f2652d2d85058cb46a5be6d5e933ebaded4b from September 12, 2017.

I notice that `jobcontrol.vim` line 16 passes `for` then `args` to `s:spawn`

```go
let job = s:spawn(a:bang, a:desc, a:for, a:args)
```

But `s:spawn` reverses that order (line 40):

```go
function! s:spawn(bang, desc, args, for) abort
```

If I switch the order to match:

```go
function! s:spawn(bang, desc, for, args) abort
```

Then things work. I can open a PR, but not sure if there's something else at play here.

**Comment by Carpetsmoker** [comment]
I think this is the same same root cause as #1444.

**Comment by bhcleek** [comment]
There's an (untested) potential fix for #1444 up now.

**Comment by Carpetsmoker** [comment]
Should be fixed with 2b3610f
